### DISCUSSION TITLE
integration with beeminder

### DISCUSSION THREAD

**Discussion by aliceinwire** [iSoron/uhabits#257] [Feature Requests]
would be nice to integrate it with beeminder

**Comment by jjst** [comment]
+1! Some more context - [beeminder](https://www.beeminder.com/overview) is a goal-tracking app with a sting. Users pledge money towards quantifiable goals they want to accomplish. If they stay on track everything's well, but if they go off-track they have to pay the pledged amount to continue tracking. Goals can either be updated manually, or automatically. Beeminder has a [public API](http://api.beeminder.com/) and integrates with services such as IFTTT, Zapier, Fitbit, Github, etc. 

The mapping between a habit and a beeminder goal seems pretty straightforward. A habit is quantifiable and has an associated goal (it has to be done `x` times in `y` days). It would be possible to [create a corresponding goal](http://api.beeminder.com/#creategoal) using the beeminder API.

I suggest adding an Integrations section in the settings with a Beeminder entry, where a user can allow the app to access their Beeminder account. If Beeminder integration is configured, then an additional setting can be displayed on the "Create habit" menu to beemind newly created habits, and potentially on the "Edit habit" menu as well to start beeminder existing habits.

**Comment by aliceinwire** [comment]
thanks for the context
